products/libllbuild/C-API.cpp

```diff
@@ -29,3 +30,10 @@ const char* llb_get_full_version_string(void) {
 int llb_get_api_version(void) {
     return LLBUILD_C_API_VERSION;
 }
+
+const char * llb_get_default_target_triple() {
+  // Use a static local to store the triple, to avoid lifetime issues.
+  static std::string tripleString = llvm::sys::getDefaultTargetTriple();
```

Comment:
does this really work? LLVM goes to some extreme lengths to determine the default triple. I am not sure if llbuild does that as well.

Reply:
I'm not quite sure what you mean. This ends up using the exact same code that would be called in the clang or Swift drivers. That code does sometimes invoke external tools or perform other heroics to determine the OS version, but all of that code looks like it's intended to be idempotent, so caching the result ought not to cause any problems.

Reply:
I meant if the copy in llbuild actually does the heroics done by llvm. I checked and it seems like this is filled in at compile time using `LLVM_DEFAULT_TARGET_TRIPLE` defined to empty string in llbuild: https://github.com/apple/swift-llbuild/blob/master/include/llvm/Config/config.h#L356

Reply:
Damn, you're right. Is there a particular reason we do this, or is it just because we haven't wanted this until now?